src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (head == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            checkIndex(index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> getNode = getNode(index);
+        Node<T> getNodePrev = getNode.prev;
+        Node<T> newNode = new Node<>(getNodePrev, value, getNode);
+        if (getNodePrev == null) {
+            head = newNode;
+        } else {
+            getNodePrev.next = newNode;
+        }
+        getNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element: list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> setNode = getNode(index);
+        T previousNodeValue = setNode.value;
+        setNode.value = value;
+        return previousNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> removeNode = getNode(index);
+        return unlink(removeNode).value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        if (node.value == object || node.value.equals(object)) {
+            unlink(node);
+            return true;
+        } else {
+            int i = 0;
+            while (node.next != null) {
+                if (node.value == object || node.value.equals(object)) {
+                    unlink(getNode(i));
```

Comment:
check null and use `node`
```suggestion
                    unlink(node);
```